optuna/visualization/_parallel_coordinate.py

```diff
@@ -167,6 +181,11 @@ def _target(t: FrozenTrial) -> float:
 
         dims.append(dim)
 
+    if len(numeric_cat_params.keys()) != 0:
+        idx = np.lexsort(list(numeric_cat_params.values())[::-1])
+        for dim in dims:
+            dim.update({"values": tuple(np.array(dim["values"])[idx])})
```

Comment:
Could you add comments to explain this logic for other maintainers? I'm afraid that I still do not fully understand this sort of logic.

Reply:
np.lexsort consumes the sort key in order from back to front. So I reversed the list to sort from head. 
(https://numpy.org/doc/stable/reference/generated/numpy.lexsort.html)
I'm going to write a comment as above, what do you think?

Reply:
Sorry, could you also explain `dim.update({"values": tuple(np.array(dim["values"])[idx])})`?